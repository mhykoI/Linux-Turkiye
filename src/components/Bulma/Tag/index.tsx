import React from 'react';
import clsx from 'clsx';
import { Bulma } from '../../../models';

export interface TagProps {
  color?: Bulma.Color;
  size?: Bulma.Size;
}

const Tag: React.FC<TagProps> = ({ color, size, children }) => {
  return (
    <span
      className={clsx(
        'tag',
        { [`is-${color}`]: !!color },
        { [`is-${size}`]: !!size },
      )}
    >
      {children}
    </span>
  );
};

export default Tag;
