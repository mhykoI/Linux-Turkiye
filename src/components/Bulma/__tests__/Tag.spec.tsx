import React from 'react';
import renderer from 'react-test-renderer';
import Tag from '../Tag';

describe('<Tag />', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(
        <Tag size="normal" color="info">
          This is a Tag!
        </Tag>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
