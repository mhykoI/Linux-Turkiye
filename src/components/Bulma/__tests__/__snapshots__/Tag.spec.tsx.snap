// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should render correctly 1`] = `
<span
  className="tag is-info is-normal"
>
  This is a Tag!
</span>
`;
